app/services/music_import_service.rb

```diff
@@ -288,7 +288,11 @@ def create_logical_structure(output, buffered_change_set_persister)
         st << { nodes: nodes, label: date }
       end
       change_set = DynamicChangeSet.new(output)
-      change_set.logical_structure = [{ label: "By Date", nodes: structure }]
+      if change_set.logical_structure[0][:label] == ["By Date"]
+        change_set.logical_structure[0].nodes += structure
```

Comment:
It's a bit confusing to use dot notation in this line but hash access in the line above.